gcc/rust/lex/rust-lex.h

```diff
@@ -136,8 +136,11 @@ class Lexer
   void split_current_token (TokenId new_left, TokenId new_right);
 
   Linemap *get_line_map () { return line_map; }
+  const char *get_filename () { return filename; }
 
 private:
+  // Original filename used to instantiate the lexer
+  const char *filename;
```

Comment:
can this be a proxy to the RAIIFile object? It would be best to decouple this and get it from the file object.

Reply:
In the end we still have to keep a `const char *` somewhere because we cannot get the filename from the `FILE` pointer kept in RAIIFile. But I agree that it makes more sense to keep it there instead of the lexer. Done in 06e6a4a25c8